contracts/multisig-prover/src/execute.rs

```diff
@@ -228,17 +233,37 @@ pub fn update_worker_set(deps: DepsMut, env: Env) -> Result<Response, ContractEr
 
             save_next_worker_set(deps.storage, &new_worker_set)?;
 
-            let mut builder = CommandBatchBuilder::new(config.destination_chain_id, config.encoder);
-            builder.add_new_worker_set(new_worker_set)?;
+            let msg_to_sign;
+            #[cfg(not(feature = "amplifier-gateway"))]
+            {
+                let mut builder =
+                    CommandBatchBuilder::new(config.destination_chain_id, config.encoder);
+                builder.add_new_worker_set(new_worker_set)?;
 
-            let batch = builder.build()?;
+                let batch = builder.build()?;
 
-            COMMANDS_BATCH.save(deps.storage, &batch.id, &batch)?;
-            REPLY_BATCH.save(deps.storage, &batch.id)?;
+                COMMANDS_BATCH.save(deps.storage, &batch.id, &batch)?;
+                REPLY_BATCH.save(deps.storage, &batch.id)?;
+
+                msg_to_sign = batch.msg_digest();
+            }
```

Comment:
Can we extract this into a function? As well as the case where the feature flag is enabled?